### PR TITLE
Bootstrap CLI package

### LINKED ISSUE #455: Bootstrap the CLI project as a hardhat plugin with a placeholder "interact" task.
Use both the v3 deployer package, and the v2x interact task as reference.

In the v2x task, we can see how we build the CLI UI, interact with the contracts etc. Keep in mind that this code is a simple hacky script, and we want things to be more robust in v3.

In the v3 deployer task, we can see how to subdivide a task in subtasks to keep code simple and robust, how to bootstrap the plugin, how to set up sample projects, etc.

In this issue, we only need to bootstrap the project with a dummy task.

Like in the deployer plugin, add a test/fixture-projects/sample-project folder with a sample router compatible project that will be used to test the plugin, both manually and programmatically through mocha.

The folder for now should just be a vanilla hardhat project that loads the deployer plugin, with a few contracts with dummy code to interact with.

No need to set up anything for tests just yet. We will be adding automated tests later. We can manually test the cli for now by importing it as a plugin in `synthetix-main`.

### PR DESCRIPTION
**Pull request by i-stam**
FIX #455

### CODE REVIEW

File: packages/synthetix-cli/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": ""
+  },
+  "author": "",
+  "license": "MIT",
+  "devDependencies": {
+    "@nomiclabs/hardhat-ethers": "^2.0.2",
+    "ethers": "^5.5.1",
+    "hardhat": "^2.6.8"
```
Comment (leomassazza) [inline packages/synthetix-cli/package.json:14]: I'd suggest using 2.6.6 as we have on the other packages and maybe move all together in a separate PR

File: packages/synthetix-cli/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": ""
+  },
+  "author": "",
+  "license": "MIT",
+  "devDependencies": {
+    "@nomiclabs/hardhat-ethers": "^2.0.2",
+    "ethers": "^5.5.1",
```
Comment (leomassazza) [inline packages/synthetix-cli/package.json:13]: ethers is included in @nomiclabs/hardhat-ethers

**Review by leomassazza — APPROVED** [review]
Approved with some np

File: packages/synthetix-cli/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
Comment (ajsantander) [inline packages/synthetix-cli/.gitignore:1]: I think we can name this package _just_ cli, so that the fully qualified package name is @synthetixio/cli

Comment (ajsantander) [reply packages/synthetix-cli/.gitignore:1]: Also, this git ignore might be redundant (see deployer package's .gitignore). The workspace ignore might be enough.

File: packages/synthetix-cli/package.json (hunk 2 above)
Comment (ajsantander) [reply packages/synthetix-cli/package.json:13]: I think it's alright, since he's following the pattern used in the deployer package. If you do apply Leo's suggestion, I think it should be on both plugins.

File: packages/synthetix-cli/tasks/interact.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const { task } = require('hardhat/config');
+
+const { TASK_INTERACT } = require('../task-names');
+
+task(TASK_INTERACT, 'Interacts with a given Synthetix deployment')
```
Comment (ajsantander) [inline packages/synthetix-cli/tasks/interact.js:5]: Description should read "Interacts with a given modular system deployment" (it's not just Synthetix now)

Also, I think it's safe to remove all params for now. Most of them don't apply anymore in the v3 architecture, and we should probably add them only as they are actually used.

**Review by ajsantander — COMMENTED** [review]
Looks good, the only critical change I want to ask is to rename the package.

**Review by mjlescano — CHANGES_REQUESTED** [review]
Please, add the `.npmrc` file as the other packages, and pin the dependencies versions on the `package.json` (e.g. change `"^1.0.1"` to `"1.0.1"`), so we are sure to not add any unwanted vulnerabilities on automatic upgrades.

File: packages/synthetix-cli/.gitignore (hunk 3 above)
Comment (i-stam) [reply packages/synthetix-cli/.gitignore]: fixed

**Review by ajsantander — APPROVED** [review]

**Review by mjlescano — APPROVED** [review]
